apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -354,10 +356,21 @@ static void setJavaBaseModule(Class<?> targetClass) throws Throwable {
      * @return a {@link ConstantCallSite} that is the target of the invokedynamic
      */
     @Nullable
-    public static ConstantCallSite bootstrap(MethodHandles.Lookup lookup,
-                                             String adviceMethodName,
-                                             MethodType adviceMethodType,
-                                             Object... args) {
+    public static ConstantCallSite bootstrap(final MethodHandles.Lookup lookup,
+                                             final String adviceMethodName,
+                                             final MethodType adviceMethodType,
+                                             final Object... args) {
+
+        // callsite resolution needs privileged access to call Class#getClassLoader() and MethodHandles$Lookup#findStatic
+        return AccessController.doPrivileged(new PrivilegedAction<ConstantCallSite>() {
```

Comment:
Can we reuse the same `PrivilegedAction` instance here? This is called very intensively at startup, so if we can avoid this overhead, that would be nice.
I think it will make the code even nicer, where the actual implementation is located within the privileged action implementation.

Reply:
we now avoid this extra allocation when SM is disabled (default)